feat: multiple sources in Audio and Video components

arthurdenner opened the pull request:
This is my take at #364 from a few months ago.
The approach is to leverage the HTML elements to make it work.

I stopped working on it as someone else was assigned but their PR is stuck for some time now.
I'm opening this as a draft because it needs to be rebased and tested in more scenarios.
I plan to get back to it soon but feel free to have a look and provide feedback in the meantime if you're interested.

JonnyBurger commented on packages/core/src/use-media-in-timeline.ts:
```diff
@@ -58,23 +58,27 @@ export const useMediaInTimeline = ({
+		const tagName = mediaType === 'audio' ? '<Audio>' : '<Video>';
+
+		if (!mediaRef.current || !hasMetadata) {
 			return;
```
We are silently returning if no metadata, and then we are not checking for `currentSrc` in the next statement. Should we swap the order of these statements?

JonnyBurger commented on packages/core/src/video/VideoForRendering.tsx:
```diff
@@ -110,7 +119,7 @@ const VideoForRenderingForwardFunction: React.ForwardRefRenderFunction<
 				fps: videoConfig.fps,
 				frame,
-				src: props.src as string,
+				src: videoRef.current.currentSrc,
```
Does it still work if the source gets changed?

We could test it by rendering 1 second with 1 video, then rendering 1 second with another video.

It's raising my eyebrow because src is not in the useEffect dependencies anymore.

JonnyBurger reviewed (changes requested):
Thanks a lot! This looks quality!

- Just a few comments that still need to be confirmed to not cause issue.
- Can we add example code to https://remotion.dev/docs/audio and https://remotion.dev/docs/video pages?
- I removed the `.skip()` from the tests, if it now fails, they need to be fixed

arthurdenner commented:
@JonnyBurger, I've updated the PR and fixed the broken tests.

However, there is still a bug in the preview that I didn't manage to solve (related to the preview) 😞 
Please read the description and check the video with a reproduction.

JonnyBurger commented:
@arthurdenner Thanks a lot! Need some time to review but will surely do in the coming week!
